shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java

```diff
@@ -0,0 +1,316 @@
+    this.context = context;
+    this.flutterJNI = flutterJNI;
+    splitInstallManager = SplitInstallManagerFactory.create(context);
+    listener = new FeatureInstallStateUpdatedListener();
+    splitInstallManager.registerListener(listener);
+    sessionIdToName = new HashMap();
+    sessionIdToLoadingUnitId = new HashMap();
+  }
+
+  public void downloadFeature(String moduleName, int loadingUnitId) {
+    if (moduleName == null) {
+      Log.e(TAG, "Dynamic feature module name was null.");
+      return;
+    }
+
+    SplitInstallRequest request = SplitInstallRequest.newBuilder().addModule(moduleName).build();
+
+    splitInstallManager
+        // Submits the request to install the module through the
+        // asynchronous startInstall() task. Your app needs to be
+        // in the foreground to submit the request.
+        .startInstall(request)
+        // Called when the install request is sent successfully. This is different than a successful
+        // install which is handled in FeatureInstallStateUpdatedListener.
+        .addOnSuccessListener(
+            sessionId -> {
+              this.sessionIdToName.put(sessionId, moduleName);
+              this.sessionIdToLoadingUnitId.put(sessionId, loadingUnitId);
+            })
+        .addOnFailureListener(
+            exception -> {
+              switch (((SplitInstallException) exception).getErrorCode()) {
+                case SplitInstallErrorCode.NETWORK_ERROR:
+                  flutterJNI.dynamicFeatureInstallFailure(
+                      moduleName,
+                      loadingUnitId,
+                      "Install of dynamic feature module \""
+                          + moduleName
+                          + "\" failed with a network error",
+                      true);
+                  break;
+                case SplitInstallErrorCode.MODULE_UNAVAILABLE:
+                  flutterJNI.dynamicFeatureInstallFailure(
+                      moduleName,
+                      loadingUnitId,
+                      "Install of dynamic feature module \""
+                          + moduleName
+                          + "\" failed as is unavailable.",
+                      false);
+                  break;
+                default:
+                  flutterJNI.dynamicFeatureInstallFailure(
+                      moduleName,
+                      loadingUnitId,
+                      "Install of dynamic feature module \""
+                          + moduleName
+                          + "\" failed with error: \""
+                          + ((SplitInstallException) exception).getErrorCode()
+                          + "\": "
+                          + ((SplitInstallException) exception).getMessage(),
+                      false);
+                  break;
+              }
+            });
+  }
+
+  public void loadAssets(@NonNull String moduleName, int loadingUnitId) {
+    try {
+      context = context.createPackageContext(context.getPackageName(), 0);
```

Comment:
Is this intentionally overwriting the context that was passed into the constructor?

Reply:
Yes. This context override ensures that the newly installed dynamic feature module is present in the context when we ask for a new asset manager.